### PR TITLE
Add timestepping loop to Poisson case

### LINKED ISSUE #11: Generalise Poisson problem to heat equation

### PR DESCRIPTION
**Pull request by Nanoseb**

### CODE REVIEW

File: src/case_setup/Poisson/poisson.f90 [hunk 1]
```diff
@@ -197,53 +205,69 @@ program poisson
...
 
-  ! Create linear solver & set options
-  call set_equation_system(par_env, b, u, M, poisson_eq)
-  call create_solver(poisson_eq, poisson_solver)
-  call set_solver_method(pressure_solver_method_name, poisson_solver)
-  call set_solver_precon(pressure_solver_precon_name, poisson_solver)
-  call solve(poisson_solver)
+    ! Create stiffness matrix
+    call create_matrix(mat_properties, M)
+    call create_vector(vec_properties, b)
```
Comment (pbartholomew08) [inline src/case_setup/Poisson/poisson.f90:232]: I think we can do this outside the time loop?

File: src/case_setup/Poisson/poisson.f90 [hunk 2]
```diff
@@ -197,53 +205,69 @@ program poisson
...
-  call axpy(-1.0_ccs_real, u_exact, u)
+    call discretise_poisson(mesh, M)
+    ! Evaluate right-hand-side vector
+    call eval_rhs(mesh, b)
 
-  err_norm = norm(u, 2) * mesh%geo%h
-  if (par_env%proc_id == par_env%root) then
-    print *, "Norm of error = ", err_norm
-  end if
+    !Call apply_timestep(mesh, u, diag, M, b)
```
Comment (pbartholomew08) [inline src/case_setup/Poisson/poisson.f90]: ?

**Review by pbartholomew08 — CHANGES_REQUESTED** [review]

File: src/case_setup/Poisson/poisson.f90 (hunk 2 above)
Comment (Nanoseb) [reply src/case_setup/Poisson/poisson.f90]: That was commented out for debugging, shouldn't have been commited, should be fixed in the next commit

File: src/case_setup/Poisson/poisson.f90 (hunk 1 above)
Comment (Nanoseb) [reply src/case_setup/Poisson/poisson.f90:232]: I get the following error at the second timestep when doing so `mat_petsc.f90(181)[0] : ERROR: changing matrix mode without updating`
So I am going to leave it there for now

**Review by pbartholomew08 — APPROVED** [review]

**Pull request merged**